pkg/component/component.go

```diff
@@ -862,36 +862,95 @@ func GetComponentType(client *occlient.Client, componentName string, application
 }
 
 // List lists components in active application
-func List(client *occlient.Client, applicationName string) (ComponentList, error) {
+func List(client *occlient.Client, applicationName string, localConfigInfo ...config.LocalConfigInfo) (ComponentList, error) {
```

Comment:
I guess the only reason why you have multiple `config.LocalConfigInfo` is to make it optional.

This is not a good approach. I think that it would be much better, to make `localConfigInfo` a pointer to the `config.LocalConfigInfo` and pass `nil` when it needs to be called without localConafigInfo.